src/bigartm/cooccurrence_dictionary.h

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017, Additive Regularization of Topic Models.
+
+//#include <cstdio>
+#include <unordered_map>
+#include <string>
+#include <iostream>
+#include <cstdlib>
+#include <fstream>
+#include <map>
+#include <vector>
+#include <sstream>
+#include <iomanip>
+
+#include "boost/algorithm/string.hpp"
+#include "boost/filesystem.hpp"
+#include "boost/utility.hpp"
+
+enum {
+  MAX_NUM_OF_BATCHES = 1000,
+  ITEMS_PER_BATCH = 10000,
+};
+
+struct Triple {
+  double cooc_value;
+  int doc_quan;
+  int second_token_id;
+};
+
+struct Pair {
+  double cooc_value;
+  int doc_quan;
+};
+
+struct cooccurrence_info {
+  double cooc_value;
+  int doc_quan, prev_doc_id;
+};
+
+using namespace std;
+using namespace boost;
+
+class CooccurrenceBatch: private boost::noncopyable {
+  struct Cell {
+    int first_token_id;
+    int num_of_triples;
+    std::vector<Triple> records;
+  };
+
+  friend class BatchManager;
+ private:
+  std::ifstream in_batch;
+  std::ofstream out_batch;
+
+  CooccurrenceBatch(int batch_num, const char filemode, const string& disk_path);
+ public:
+  Cell cell;
+
+  void FormNewCell(std::map<int, std::map<int, cooccurrence_info>>::iterator& map_node);
+
+  int ReadCellHeader();
+
+  void ReadRecords();
+
+  int ReadCell();
+
+  void WriteCell();
+};
+
+class BatchManager {
+ private:
+  int batch_quan;
+  const int max_batch_quan = 1000;
```

Comment:
firstly declare public methods, than private methods, than public fields (it's better to declare them private with getters), than private fields